edk2: add custom patch for bootorder support in CSM mode

mihalicyn opened the pull request:
```
From ("edk2: force CSM boot mode for bootorder") description: 
========
If "-fw_cfg name=opt/com.canonical.lxd/force_csm,string=yes" is specified in the Qemu commandline then treat bootorder settings from Qemu as settings for CSM boot. Right now bootorder from Qemu is always considered as UEFI and CSM boot is always last in the priority list. Even worser, you can't really make CSM to be the first in the bootorder, because in the SetBootOrderFromQemu() function we always fixup bootorder in accordance with Qemu provided and setting from NVRAM are always got reset.

Fortunately, we don't need to patch SeaBIOS, because it already handles bootorder from Qemu properly. So, we just need to pass execution from UEFI to SeaBIOS and from this point everything will work just fine.

This is a very hacky solution but we have no choice here. Because the alternative here is to extend Qemu fwcfg "bootorder" interface to make it aware of CSM/UEFI boot modes and then patch UEFI and SeaBIOS to handle all of that. This can be an ready-to-upstream solution, but I'm not sure that it can be interesting for edk2 folks, because all of this CSM stuff considered as a legacy thing.

This thing is enabled only when CSM_ENABLE=TRUE AND "-fw_cfg name=opt/com.canonical.lxd/force_csm,string=yes" specified in the Qemu cmdline.
========
```

Fixes https://github.com/canonical/lxd/issues/11908
Related to https://github.com/canonical/lxd/pull/12564

mihalicyn commented:
Let's wait until:
https://github.com/canonical/lxd/pull/12522
https://github.com/canonical/lxd/pull/12515
will be merged

simondeziel commented on patches/edk2-0007-force-CSM-boot-mode-for-bootorder.patch:
```diff
@@ -0,0 +1,115 @@
++      if ((DevicePathType (ActiveOption[Idx].BootOption->FilePath) == BBS_DEVICE_PATH) &&
++          (DevicePathSubType (ActiveOption[Idx].BootOption->FilePath) == BBS_BBS_DP))
++      {
++        DEBUG ((DEBUG_VERBOSE, "%a: Found a legacy option: %s\n", __func__, ActiveOption[Idx].BootOption->Description));
```
```suggestion
+        DEBUG ((DEBUG_VERBOSE, "%a: Found a legacy boot option: %s\n", __func__, ActiveOption[Idx].BootOption->Description));

```

mihalicyn replied:
Fixed, thanks!

simondeziel reviewed (approved):
LGTM for the extent I could review. There are a few minor English mistakes in the patch description which is probably fine as this won't be upstream'ed.

mihalicyn committed:
edk2: add custom patch for bootorder support in CSM mode

From ("edk2: force CSM boot mode for bootorder") description:
========
If "-fw_cfg name=opt/com.canonical.lxd/force_csm,string=yes" is specified
in the Qemu commandline then treat bootorder settings from Qemu
as settings for CSM boot. Right now bootorder from Qemu is always considered
as UEFI and CSM boot is always last in the priority list. Even worser,
you can't really make CSM to be the first in the bootorder, because in the
SetBootOrderFromQemu() function we always fixup bootorder in accordance with
Qemu provided and setting from NVRAM are always got reset.

Fortunately, we don't need to patch SeaBIOS, because it already handles
bootorder from Qemu properly. So, we just need to pass execution from
UEFI to SeaBIOS and from this point everything will work just fine.

This is a very hacky solution but we have no choice here. Because
the alternative here is to extend Qemu fwcfg "bootorder" interface
to make it aware of CSM/UEFI boot modes and then patch UEFI and SeaBIOS
to handle all of that. This can be an ready-to-upstream solution,
but I'm not sure that it can be interesting for edk2 folks, because
all of this CSM stuff considered as a legacy thing.

This thing is enabled only when CSM_ENABLE=TRUE AND
"-fw_cfg name=opt/com.canonical.lxd/force_csm,string=yes"
specified in the Qemu cmdline.
========

Signed-off-by: Alexander Mikhalitsyn <aleksandr.mikhalitsyn@canonical.com>

tomponline reviewed (approved):
LGTM thanks!
